### PR TITLE
Add guide for shipping alternative platforms

### PR DESCRIPTION
**Pull request by saschagrunert**
This outline should help contributors to get an idea about what needs to be done to add additional supported platforms to Kubernetes.

#### Which issue(s) this PR fixes

Supersedes https://github.com/kubernetes/community/pull/5014

#### Main changes to the initial version:

- Went from `architectures` to `platforms`, which should not cover operating systems as well.
- Removed the `TODO` list since I think this is now covered
- Addressed previous review comments but tried to be not too specific for this short abstract.

I tried to not change too much of the document to still preserve the outlined style.

PTAL @dims @justaugustus @tpepper @LappleApple

### CODE REVIEW

**Comment by saschagrunert** [comment]
We may add a cross reference to this doc from one of the other documents.

**Comment by dims** [comment]
@saschagrunert sounds good. Please drop a note to arch mailing list asking for reviews and starting a clock. We can merge this after next sig-arch meeting (please ping @johnbelamaric  to add this to agenda).

/assign @johnbelamaric @derekwaynecarr @liggitt

File: contributors/devel/sig-architecture/platforms.md [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+platforms. Specifically to call out the ability to run conformance tests on
+these platforms.
+
+Target of this document is to provide a starting point for adding new platforms
+to Kubernetes from a SIG Architecture perspective. This does not include release
+mechanics or supportability in terms of functionality.
+
+# Step 1: crawling (Build)
+
+- bazel based build infrastructure should support this architecture
```
Comment (aojea) [inline contributors/devel/sig-architecture/platforms.md:24]: `bazel based ...` this is going to be funny :) ;)

File: contributors/devel/sig-architecture/platforms.md [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+- docker based build infrastructure should support this architecture
+
+The above 2 implicitly require the following:
+
+- golang should support the architecture out-of-the-box.
+- All our dependencies, whether vendored or run separately, should support this
+  architecture out-of-the-box.
```
Comment (aojea) [inline contributors/devel/sig-architecture/platforms.md:31]: nit, you mentioned

> Went from architectures to platforms, which should not cover operating systems as well.

maybe is a leftover `architecture` here?

Comment (saschagrunert) [reply contributors/devel/sig-architecture/platforms.md]: Yep, changed it to `platform`, thanks :+1:

File: contributors/devel/sig-architecture/platforms.md (hunk 1 above)
Comment (mkumatag) [reply contributors/devel/sig-architecture/platforms.md:24]: I second that! bazel on non-amd64 architectures will always be challenging.

Comment (saschagrunert) [reply contributors/devel/sig-architecture/platforms.md:24]: Right now we build the binary artifacts via `make cross-in-a-container` which uses the kube-cross container image, which does not seem to use bazel.

Comment (dims) [reply contributors/devel/sig-architecture/platforms.md:24]: that's just the release process. regular folks should be still able to use bazel like they do for development workflows. (UNTIL we get of bazel entirely)

Comment (saschagrunert) [reply contributors/devel/sig-architecture/platforms.md:24]: Agreed :+1:

File: contributors/devel/sig-architecture/platforms.md [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+- Once Step 3 is done, binaries / images in arch can ship with release.
+- Step 2 is at least the default e2e-gce equivalent, PLUS the node e2e tests.
+  More the better.
+- Step 2 will involve 3rd party reporting to test-grid at the least.
+- Step 2 may end up needing boskos etc to run against clouds (with these arches)
+  where we have credits:
+- Step 3 is at least the conformance test suite. More the better. Using
+  community tools like prow/kubeadm is encouraged but not mandated.
+- Step 4 is where we take this up to CNCF trademark program. Need things solid
+  for at least a year in Step 3 before we go to Step 4.
```
Comment (johnbelamaric) [inline contributors/devel/sig-architecture/platforms.md:127]: I put this on the agenda for the sig-arch meeting next week.

I don't see how we can control Step 4 at all, assuming the vendor is able to build the conformance test suite and images and run the conformance tests successfully. That is, I am not sure that integration into our OSS build, test, and release process are strictly necessary for someone to release k8s running on a new architecture, although it sure would make it a lot easier for them.

File: contributors/devel/sig-architecture/platforms.md (hunk 1 above)
Comment (vielmetti) [reply contributors/devel/sig-architecture/platforms.md:24]: Not only does the Bazel binary need to support any new architecture, but also the various and sundry rules files need to work as well. (Currently tracking a specific rules_docker dependency for arm64, for example, which is blocking `kind`.) See https://github.com/kubernetes-sigs/kind/issues/166

File: contributors/devel/sig-architecture/platforms.md (hunk 3 above)
Comment (derekwaynecarr) [reply contributors/devel/sig-architecture/platforms.md:127]: i agree with @johnbelamaric .

File: contributors/devel/sig-architecture/platforms.md (hunk 1 above)
Comment (derekwaynecarr) [reply contributors/devel/sig-architecture/platforms.md:24]: i do not see why bazel is a strong requirement.
is the language 'should' implying MUST or MAY?

Comment (cwsolee) [reply contributors/devel/sig-architecture/platforms.md:24]: Bazel on s390x is quite problematic too. Looks like it applies to other arch too.

File: contributors/devel/sig-architecture/platforms.md (hunk 3 above)
Comment (cwsolee) [reply contributors/devel/sig-architecture/platforms.md:127]: Looks fair to me(s390x).

File: contributors/devel/sig-architecture/platforms.md (hunk 1 above)
Comment (saschagrunert) [reply contributors/devel/sig-architecture/platforms.md:24]: Removed the bazel restriction :heavy_check_mark:

**Pull request merged**